float8_experimental/float8_linear.py

```diff
@@ -232,7 +232,7 @@ def cast_w_to_float8(
             self.fp8_scale_w,
             scale_fn_name,
             torch.float8_e4m3fn,
-            is_amax_initialized,
+            self.is_amax_initialized,
```

Comment:
We have to push this `is_amax_initialized` into the cast function since the `Float8LinearWeightTensor` subclass cannot keep a reference to the bool.